### PR TITLE
f-analytics@0.9.0 - Moved push 'UserData' from mixin to service

### PR DESCRIPTION
**Pull request by oliversweb**
### Changed
- Moved the prepUserData from Mixin into Service to allow the consumer to dictate when to push the GA data.

### CODE REVIEW

File: packages/services/f-analytics/README.md [hunk 1]
```diff
@@ -23,9 +23,11 @@ This component abstracts away the gathering of the various data values needed fo
 - Single point to record GA data: Currently we have GA/GTM logic scattered throught various features, this will allow that logic to be removed and centralise it a single component.
 - Self-sufficient: With only supplying a small amount of global data this component will attempt to evaluate, gather and record all the data required for the GA `platformData`, `userData` and `pageData` models.
 - Provide the facility to push 'ad-hoc' GA events via a global service (even if serverside).
+- Provide the facility to push 'ad-hoc' GA events via a global service (even if serverside).
```
Comment (dandel10n) [inline packages/services/f-analytics/README.md]: This line duplicates the line above it

File: packages/services/f-analytics/README.md [hunk 2]
```diff
@@ -23,9 +23,11 @@ This component abstracts away the gathering of the various data values needed fo
...
 
 ## Benefits (Soon)
 - _extend the data properties for the GA `platformData` model_
+- _evaluate, gather and produce data for the GA `pageData` model_
```
Comment (dandel10n) [inline packages/services/f-analytics/README.md]: I think this is already in current benefits

File: packages/services/f-analytics/src/mixins/_tests/analytics.methods.test.js [hunk 3]
```diff
@@ -13,50 +12,6 @@ import analyticsMixin from '../analytics.mixin.vue';
...
-const userDataWithoutMockAuthToken = {
-    'a-UserId': userIdFromCookie,
-    authType: undefined,
-    email: undefined,
-    globalUserId: undefined,
-    signinType: undefined,
-    signupDate: undefined
-};
-
 const mockAuthToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.'
```
Comment (dandel10n) [inline packages/services/f-analytics/src/mixins/_tests/analytics.methods.test.js]: This `mockAuthToken` is the same as `mockAuthTokenRegistered` as I believe Andy copied it from my userData PR for the pageData test.

File: packages/services/f-analytics/src/plugins/analytics.plugin.js [hunk 4]
```diff
@@ -1,43 +1,23 @@
...
-        if (process.env.justEatEnvironment) platformData.environment = process.env.justEatEnvironment;
-        if (process.env.FEATURE_VERSION) platformData.version = process.env.FEATURE_VERSION;
-        if (process.env.INSTANCE_POSITION) platformData.instancePosition = process.env.INSTANCE_POSITION;
-
-        const userPercent = getCookie('je-user_percentage', req);
-        if (userPercent) platformData.jeUserPercentage = userPercent;
+        mapServersidePlatformData(platformData, req);
 
-        store.dispatch(`${options.namespace}/${UPDATE_PLATFORM_DATA}`, platformData);
+        store.dispatch(`${options.namespace}/updatePlatformData`, platformData);
```
Comment (dandel10n) [inline packages/services/f-analytics/src/plugins/analytics.plugin.js:14]: What is the reason for this change?

Comment (oliversweb) [reply packages/services/f-analytics/src/plugins/analytics.plugin.js:14]: Because the these constants are for Mutations only and I was hijacking them for actions as well :\ ... so reverted to text.

File: packages/services/f-analytics/src/services/analytics.mapper.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import SHA256 from 'crypto-js/sha256';
+import {
+    COUNTRY_INFO,
+    DEFAULT_APP_TYPE,
+    DEFAULT_APP_ID,
+    IDENTITY_PROVIDERS,
+    GRANT_TYPES
+} from '@/constants';
+
+const getCookie = (name, req) => {
```
Comment (kevinrodrigues) [inline packages/services/f-analytics/src/services/analytics.mapper.js:18]: Can you add some JSdoc comments to these methods?

File: packages/services/f-analytics/src/services/analytics.mapper.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+const mapAnonymousUserId = (userData, req) => {
+    const value = getCookie('je-auser', req);
+    if (value) {
+        userData['a-UserId'] = value;
+    }
+};
+
+const mapUserAgent = (platformData, req) => {
+    let userAgentString;
+    if (typeof (window) !== 'undefined' && navigator) {
```
Comment (zlatinivanov-je) [inline packages/services/f-analytics/src/services/analytics.mapper.js]: what is the reason for the syntax of this check? `typeof (window) !== 'undefined'` 
Would `window && window.navigator` or even `window?.navigator` not be enough? 
I would suggest prefixing `navigator` with `window` everywhere to avoid confusion with possibly other Object `navigator`

**Review by zlatinivanov-je — DISMISSED** [review]
One small comment and like the suggestion of adding JSDoc comments for methods
Other than that LGTM

**Review by zlatinivanov-je — APPROVED** [review]

**Review by dandel10n — APPROVED** [review]
